### PR TITLE
fix(client): Make Font Awesome Icons accessible to visually impaired campers

### LINKED ISSUE #15739: Make Font Awesome Icons accessible to visually impaired campers
A lot of campers who are working through freeCodeCamp are visually impaired or completely blind. We should use additional HTML elements everywhere where we use a Font Awesome icon that explains what the icon means.

Font Awesome has a guide for this here: http://fontawesome.io/accessibility/

### PR DESCRIPTION
**Pull request by xnt**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #15739

Adds `aria-label` tags that explain the icons. Note that I also had to explicitly add `aria-hidden="false"`, probably because of https://github.com/FortAwesome/react-fontawesome/issues/126 .

Interestingly, when using the `ariaLabel` prop notation, the DOM attribute would be rendered verbatim 🤷‍♂️ . So I was explicit on it as well.

I also thought that the labels should represent the icon itself, and not the link/text they're related to (since the text would be read as well, resulting in duplication).

### CODE REVIEW

**Review by ValeraS — CHANGES_REQUESTED** [review]
@xnt Please check a Font Awesome guide here https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility and amend appropriately.

**Comment by xnt** [comment]
@ValeraS what would you like to see specifically? Titles instead of aria-label ?

File: client/src/components/profile/components/SocialIcons.js [hunk 1]
```diff
@@ -27,31 +27,51 @@ const propTypes = {
 function LinkedInIcon(linkedIn) {
   return (
     <a href={linkedIn} rel='noopener noreferrer' target='_blank'>
-      <FontAwesomeIcon icon={faLinkedin} size='2x' />
+      <FontAwesomeIcon
+        aria-hidden='false'
+        aria-label='LinkedIn icon'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: Aria label should be set on a link tag and describe where it goes. Additional attributes on icon are not need.

File: client/src/components/profile/components/SocialIcons.js [hunk 2]
```diff
@@ -27,31 +27,51 @@ const propTypes = {
...
   );
 }
 
 function GithubIcon(ghURL) {
   return (
     <a href={ghURL} rel='noopener noreferrer' target='_blank'>
-      <FontAwesomeIcon icon={faGithub} size='2x' />
+      <FontAwesomeIcon
+        aria-hidden='false'
+        aria-label='GitHub icon'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: The same as above.

File: client/src/components/profile/components/SocialIcons.js [hunk 3]
```diff
@@ -27,31 +27,51 @@ const propTypes = {
...
   );
 }
 
 function WebsiteIcon(website) {
   return (
     <a href={website} rel='noopener noreferrer' target='_blank'>
-      <FontAwesomeIcon icon={faLink} size='2x' />
+      <FontAwesomeIcon
+        aria-hidden='false'
+        aria-label='Link icon'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: The same as above

File: client/src/components/profile/components/SocialIcons.js [hunk 4]
```diff
@@ -27,31 +27,51 @@ const propTypes = {
...
   );
 }
 
 function TwitterIcon(handle) {
   return (
     <a href={handle} rel='noopener noreferrer' target='_blank'>
-      <FontAwesomeIcon icon={faTwitter} size='2x' />
+      <FontAwesomeIcon
+        aria-hidden='false'
+        aria-label='Twitter icon'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: The same as above

**Review by ValeraS — CHANGES_REQUESTED** [review]
All other icons are decorative and should be staying hidden from screen readers.

File: client/src/components/profile/components/SocialIcons.js (hunk 1 above)
Comment (xnt) [reply client/src/components/profile/components/SocialIcons.js]: Done

File: client/src/components/profile/components/SocialIcons.js (hunk 3 above)
Comment (xnt) [reply client/src/components/profile/components/SocialIcons.js]: Done

File: client/src/components/profile/components/SocialIcons.js (hunk 4 above)
Comment (xnt) [reply client/src/components/profile/components/SocialIcons.js]: Done

**Comment by xnt** [comment]
@ValeraS changes applied, thank you for the detail 🙇 !!

I had mixed feelings regarding `Link to {website}` versus `Link to our website`. Let me know if you'd like me to change this.

Cheers and thanks for your time!

**Review by xnt — COMMENTED** [review]
Changes implemented, thank you!

File: client/src/pages/index.js [hunk 5]
```diff
@@ -153,11 +153,7 @@ const IndexPage = () => (
           <h2 className='black-text'>HTML5</h2>
         </Col>
         <Col md={3} sm={6} xs={12}>
-          <FontAwesomeIcon
-            className='landing-skill-icon'
-            icon={faCss3Alt}
-            size='9x'
-          />
+          <FontAwesomeIcon icon={faCss3Alt} size='9x' />
```
Comment (ValeraS) [inline client/src/pages/index.js]: Revert this changes, please.

File: client/src/components/profile/components/SocialIcons.js [hunk 6]
```diff
@@ -26,31 +26,51 @@ const propTypes = {
 
 function LinkedInIcon(linkedIn) {
   return (
-    <a href={linkedIn} rel='noopener noreferrer' target='_blank'>
+    <a
+      aria-label='Link to our LinkedIn'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: It's a link to a camper's LinkedIn page. I think it should be something like this:
 ```suggestion
       aria-label=`Link to ${username}'s LinkedIn`
```

`username` you can pass in `props` from a Camper component.

File: client/src/components/profile/components/SocialIcons.js [hunk 7]
```diff
@@ -26,31 +26,51 @@ const propTypes = {
...
       <FontAwesomeIcon icon={faLinkedin} size='2x' />
     </a>
   );
 }
 
 function GithubIcon(ghURL) {
   return (
-    <a href={ghURL} rel='noopener noreferrer' target='_blank'>
+    <a
+      aria-label='Link to our Facebook'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: the same

File: client/src/components/profile/components/SocialIcons.js [hunk 8]
```diff
@@ -26,31 +26,51 @@ const propTypes = {
...
       <FontAwesomeIcon icon={faGithub} size='2x' />
     </a>
   );
 }
 
 function WebsiteIcon(website) {
   return (
-    <a href={website} rel='noopener noreferrer' target='_blank'>
+    <a
+      aria-label={`Link to ${website}`}
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: the same

File: client/src/components/profile/components/SocialIcons.js [hunk 9]
```diff
@@ -26,31 +26,51 @@ const propTypes = {
...
       <FontAwesomeIcon icon={faLink} size='2x' />
     </a>
   );
 }
 
 function TwitterIcon(handle) {
   return (
-    <a href={handle} rel='noopener noreferrer' target='_blank'>
+    <a
+      aria-label='Link to our Twitter'
```
Comment (ValeraS) [inline client/src/components/profile/components/SocialIcons.js]: the same

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: client/src/components/profile/components/SocialIcons.js (hunk 6 above)
Comment (xnt) [reply client/src/components/profile/components/SocialIcons.js]: I see, thank you! I thought they were FCC's social icons. I will implement the changes you requested when I have a bit of downtime.

Totally appreciate the time you're investing on this 🙇 !! Apologies for the back-and-forth.

**Review by xnt — COMMENTED** [review]
Fixed and tested locally, thank you!

**Review by ValeraS — APPROVED** [review]

**Comment by ValeraS** [comment]
@xnt Thanks for your contribution.

**Pull request merged**